packages/fcl/src/current-user/index.js

```diff
@@ -93,44 +93,62 @@ function notExpired(user) {
   )
 }
 
+async function makeMessage() {
+  return {
+    timestamp: Date.now(),
+    appDomainTag: await config.get("fcl.appDomainTag"),
+    extensions: window.fcl_extensions || [],
+  }
+}
+
 async function authenticate({service, redir = false}) {
   return new Promise(async (resolve, reject) => {
     spawnCurrentUser()
     const user = await snapshot()
-    if (user.loggedIn) return resolve(user)
-
-    const {discoveryWallet, discoveryWalletMethod, appDomainTag} =
-      await buildAuthnConfig()
+    const msg = await makeMessage()
+    const opts = {redir}
+    const discoveryService = await getDiscoveryService()
```

Comment:
Heads up we may get a small merge conflict here.

Reply:
yeah, ready for that TY